nsqd/nsqd.go

```diff
@@ -206,6 +212,30 @@ func (n *NSQD) GetStartTime() time.Time {
 	return n.startTime
 }
 
+// AddClient adds a client to the producers map
+func (n *NSQD) AddClient(clientID int64, client Producer) {
+	n.Lock()
+	defer n.Unlock()
```

Comment:
if we're really going to have to call `AddClient()` on every single `PUB`, we can make it a bit more efficient by not using `defer`